### PR TITLE
adds user authentication

### PR DESCRIPTION
**Pull request by dskamiotis**
Adds user authentication via google cloud services API to authenticate the user has and is a valid Guardian domain extension email.

### CODE REVIEW

File: app/config/Config.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+class Config(context: ApplicationLoader.Context) {
+  val isDev: Boolean = context.environment.mode == Mode.Dev
+  lazy val no2faUser: String = "composer.test@guardian.co.uk"
+  lazy val domain: String = "local.dev-gutools.co.uk"
+  lazy val host: String = s"https://atom-preview.$domain"
+  lazy val pandaSystem: String = "workflow"
+  lazy val pandaBucketName: String = "pan-domain-auth-settings"
+  lazy val pandaSettingsFile: String = s"$domain.settings"
+  lazy val credentialsProvider = new AWSCredentialsProviderChain(
+    new ProfileCredentialsProvider("workflow"),
```
Comment (akash1810) [inline app/config/Config.scala:18]: As this app will run in the Composer account, should we use Composer credentials?

File: app/controllers/PanDomainAuthActions.scala [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import com.gu.pandomainauth.model.AuthenticatedUser
+import config.Config
+import play.api.{Logger, Logging}
+import play.api.mvc._
+
+trait PanDomainAuthActions extends AuthActions with Results with Logging {
+  def config: Config
+
+  override def validateUser(authedUser: AuthenticatedUser): Boolean = {
+    (authedUser.user.emailDomain == "guardian.co.uk") &&
```
Comment (akash1810) [inline app/controllers/PanDomainAuthActions.scala:13]: Worth moving the `guardian.co.uk` string into Config?

File: app/controllers/PanDomainAuthActions.scala [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    logger.info(message)
+    Ok((message))
+  }
+
+  override def invalidUserMessage(claimedAuth: AuthenticatedUser): String = {
+    if( (claimedAuth.user.emailDomain == "guardian.co.uk") && !claimedAuth.multiFactor) {
+      s"${claimedAuth.user.email} is not valid for use with Workflow as you need to have two factor authentication enabled." +
+        s" Please contact the Helpdesk by emailing 34444@theguardian.com or calling 34444 and request access to Composer CMS tools"
+    } else {
+      s"${claimedAuth.user.email} is not valid for use with Workflow. You need to use your Guardian Google account to login. Please sign in with your Guardian Google account first, then retry logging in"
```
Comment (akash1810) [inline app/controllers/PanDomainAuthActions.scala:29]: Shall we replace "Workflow" here with `config.pandaSystem`?

**Review by akash1810 — APPROVED** [review]
LGTM! Added a couple of, non-blocking, minor comments.

File: app/controllers/PanDomainAuthActions.scala (hunk 3 above)
Comment (dskamiotis) [reply app/controllers/PanDomainAuthActions.scala:29]: Yes great idea 💡 !

**Comment by davidfurey** [comment]
@dskamiotis this PR hasn't deployed successfully so I have rolled back to the last good deploy (build 48)

**Pull request merged**